apis/cr/v1/deepcopy.go

```diff
@@ -87,10 +87,23 @@ func (in *Pgcluster) DeepCopyInto(out *Pgcluster) {
 		UserSecretName:     in.Spec.UserSecretName,
 		RootSecretName:     in.Spec.RootSecretName,
 		PrimarySecretName:  in.Spec.PrimarySecretName,
+		CollectSecretName:  in.Spec.CollectSecretName,
```

Comment:
Its worth noting that the deep copu functions are typically auto-generated using the k8s code generator: https://github.com/kubernetes/code-generator, and therefore are not typically modified manually.

As part of an upcoming change I will be regenerating all of our deep copy functions, which should hopefully still provide you with the function you require here.  Therefore just making a note so that we can keep than in mind here.

Reply:
ok, cool, that sounds good to me. For now, I'll just remove the currentprimary entry, as it will no longer be used regardless.